### PR TITLE
Add Intro to R for RNA-Seq Analysis

### PR DESCRIPTION
**Pull request by raynamharris**
## PR Checklist

- [x] Release label added
- [x] PR category label added
- [x] PR mergeable
- [x] Clean build logs
- [x] Linked related issues #532
- [x] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description

Adds the R lesson from recent workshops and a hackathon

**Preview link**

https://cfde-training-and-engagement--535.com.readthedocs.build/General-Tools/R-for-RNA-Seq/

## Review format

Please check

- [x] spelling
- [x] hyperlinks
- [x] code formatting
- [x] readability

## Timeline

For June 15 release

### CODE REVIEW

**Comment by jeremywalter** [comment]
Things that might need attention:
visualize.md --> in the `geom_point() `section there are a series of plots, the first two are explicitly explained (initial plot and inverse log)... three more plots follow that aren't captioned/explained in the text.

File: docs/General-Tools/R-for-RNA-Seq/import.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
+---
+title: Import Data
+---
+
+Data can be imported using functions from `base` R (such as `read.csv()` and 
+`read.table()`) or with with functions from `readr`(such as `read_csv()` and `read_tsv()`).
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/import.md]:
```suggestion
`read.table()`) or with functions from `readr`(such as `read_csv()` and `read_tsv()`). 
```

File: docs/General-Tools/R-for-RNA-Seq/import.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
+---
+title: Import Data
+---
+
+Data can be imported using functions from `base` R (such as `read.csv()` and 
+`read.table()`) or with with functions from `readr`(such as `read_csv()` and `read_tsv()`). 
+There are subtle differences in the default behavior these function, included how they treat dashes and spaces in column names, whether headers and row names are default. For this workshop, we will use `read.csv()` and `read.table()`.
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/import.md]:
```suggestion
There are subtle differences in the default behavior these of functions, including how they treat dashes and spaces in column names, whether headers and row names are default. For this workshop, we will use `read.csv()` and `read.table()`. 
```

File: docs/General-Tools/R-for-RNA-Seq/.pages [hunk 3]
```diff
@@ -0,0 +1,7 @@
+nav:
+- index.md
+- intro.md 
+- import.md
+- visualize.md
+- wrangle.md
+- wrapup.md
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/.pages]: I see a warning here there is no new line at the end of this file. if that is intentional then ignore this comment!

File: docs/General-Tools/R-for-RNA-Seq/intro.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    ## [1] 202
+
+``` r
+log10(0.05)
+```
+
+    ## [1] -1.30103
+
+You can save variable and recall them later.
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/intro.md]:
```suggestion
You can save variables and recall them later.
```

File: docs/General-Tools/R-for-RNA-Seq/visualize.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
+---
+title: Visualize Data
+---
+
+### `ggplot()`
+
+`ggplot2` is a very popular package for making visualization. It is
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/visualize.md]:
```suggestion
`ggplot2` is a very popular package for making visualizations. It is
```

File: docs/General-Tools/R-for-RNA-Seq/visualize.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+built on the “grammar of graphics”. Any plot can be expressed from the
+same set of components: a data set, a coordinate system, and a set of
+“geoms” or the visual representation of data points such as points,
+bars, line, or boxes. This is the template we build on:
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/visualize.md]:
```suggestion
bars, lines, or boxes. This is the template we build on:
```

File: docs/General-Tools/R-for-RNA-Seq/intro.md (hunk 4 above)
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/intro.md]:
```suggestion
You can save variables and recall them later.
```

File: docs/General-Tools/R-for-RNA-Seq/visualize.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+```r
+head(samples)
+ggplot(samples, aes(x = SMTS, color = AGE)) +
+geom_bar(stat = "count") + 
+coord_flip()
+```
+               
+![](./images/bar3-1.png)
+
+Note that the bars are outlined in a color according to hardy scale. If
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/visualize.md]: should this be "according to the Hardy scale."? not sure "the" is grammatically needed and if Hardy should be capitalized, so I'm not making a suggestion. (ignore if this is wrong)

File: docs/General-Tools/R-for-RNA-Seq/wrangle.md [hunk 8]
```diff
@@ -0,0 +1,621 @@
...
+    ## ACTA1    -1.358451 10.437939 -2.446102 0.016765666 0.10888918 -3.1289828
+    ## ADAMTSL2  1.338257  5.208146  2.900833 0.004871530 0.06245142 -2.2916348
+    ## ADH1B     1.259668  7.381462  3.382176 0.001141426 0.03624785 -0.9658612
+    ## ADIPOQ   -1.119484  1.117207 -1.720643 0.089405972 0.26371302 -4.3157948
+    ## AJAP1     1.010117 -1.212201  2.790425 0.006659281 0.07018819 -2.4766685
+    ## ALAS2     1.122494 -1.039829  1.840601 0.069603628 0.22901993 -4.0459220
+
+### `arrange()`
+
+Sometimes its nice to arrange by p-value. By default, the `arrange()` function will sort characters alphabetically and numbers in ascending order. Use `arange(desc())` to sort in the reverse order.
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/wrangle.md]:
```suggestion
Sometimes it's nice to arrange by p-value. By default, the `arrange()` function will sort characters alphabetically and numbers in ascending order. Use `arange(desc())` to sort in the reverse order. 
```

File: docs/General-Tools/R-for-RNA-Seq/wrangle.md [hunk 9]
```diff
@@ -0,0 +1,621 @@
...
+    ## [26] "CSRP3"        "FZD10-AS1"    "LINC02268"    "GDF15"        "PHF21B"      
+    ## [31] "CPXM1"        "IL24"         "ADH1B"        "MCF2"         "WWC1"        
+    ## [36] "SGPP2"        "COL24A1"      "SEC24AP1"     "ANKRD1"       "CDO1"        
+    ## [41] "CCL28"        "SLC5A10"      "XIRP2"
+
+=== "Challenge"
+
+    Replace the input results file with a different file, such as the
+    results of the comparison of 20-29 and 50-59 year old heart samples.
+    What are the deferentially expressed genes?
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/wrangle.md]:
```suggestion
    What are the differentially expressed genes?
```

File: docs/General-Tools/R-for-RNA-Seq/wrangle.md [hunk 10]
```diff
@@ -0,0 +1,621 @@
...
+
+Most RNA-Seq pipelines require that the counts file to be in a matrix
+format where each sample is a column and each gene is a row and all the
+values are integers or doubles with all the experimental factors in a
+separate file. Moreover, we need a corresponding file where the row
+names are the sample id and they match the column names of the counts
+file.
+
+When you type `rownames(colData) == colnames(counts)` you should see
+many TRUE statments. If the answer if FALSE your data cannot be
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/wrangle.md]:
```suggestion
many TRUE statements. If the answer if FALSE your data cannot be
```

File: docs/General-Tools/R-for-RNA-Seq/wrangle.md [hunk 11]
```diff
@@ -0,0 +1,621 @@
...
+### `pivot_longer()`
+
+The matrix form of the count data is required for some pipelines, but
+many R programs are better suited to data in a long format where each
+row is an observation. I like to create `counts_tidy_long` file that can
+be easily subset by variables or genes of interest.
+
+Because the count files are so large, it is good to filter the counts
+first. I’ll filter by `rowSums(.) > 0` and then take the top 6 with
+`head()`. Then crate a column for lengthening.
```
Comment (jessicalumian) [inline docs/General-Tools/R-for-RNA-Seq/wrangle.md]:
```suggestion
`head()`. Then create a column for lengthening.
```

**Review by jessicalumian — APPROVED** [review]
looks good! I fixed typos with suggested changes

File: docs/General-Tools/R-for-RNA-Seq/.pages (hunk 3 above)
Comment (raynamharris) [reply docs/General-Tools/R-for-RNA-Seq/.pages]: hmm will look into this. thanks :)

Comment (ctb) [reply docs/General-Tools/R-for-RNA-Seq/.pages]: (I don't think it matters.)

File: docs/General-Tools/R-for-RNA-Seq/visualize.md (hunk 7 above)
Comment (raynamharris) [reply docs/General-Tools/R-for-RNA-Seq/visualize.md]: fixed

**Comment by raynamharris** [comment]
Screen Shot 2022-06-23 at 10 29 22 AM

**Pull request merged**